modules/launcher/src/main/java/org/mule/runtime/module/launcher/plugin/ArtifactPluginDescriptorLoader.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.launcher.plugin;
+
+import static org.apache.commons.lang.StringUtils.removeEnd;
+import static org.mule.runtime.core.util.FileUtils.unzip;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Loads a {@link ArtifactPluginDescriptor} from a file resource.
+ *
+ * @since 4.0
+ */
+public class ArtifactPluginDescriptorLoader
```

Comment:
There should be a test fro this, probably moved from the old test classes and use a mock in the old ones